WordPress/gutenberg
# Fix exhaustive deps warning

## What?
This adds all the dependencies to this useEffect.

## Why?
We tried this once before and it had unexpected side-effects, so we [reverted it](https://github.com/WordPress/gutenberg/pull/48599). The side-effects were fixed in https://github.com/WordPress/gutenberg/pull/48219, so we should be good to add this dependency back in.

## How?
Just adds the dependency to the dependency array.

## Testing Instructions
1. Switch to a theme that has inner blocks set inside the navigation block, or add this to a navigation block:
```
		
			
		
```
2. Open the Site Editor
3. Select the navigation block
4. Add a block to the navigation (either in the list view or in the canvas)
5. Check that you see a notification saying that a new navigation has been created
6. Check that you only have one navigation in `wp-admin/edit.php?post_type=wp_navigation`
